### DISCUSSION TITLE
support multiple php version

### DISCUSSION THREAD

**Discussion by wnunezc** [leokhoa/laragon#990] [Laragon's Articles & How-tos]
i have use multiple version of php at same time runing...

in xamp i follow this tutorial: https://themanish.com/run-multiple-version-of-php-in-xampp-windows-without-installing-multiple-xampp/

Any way to get the same in laragon??

**Comment by anburocky3** [comment]
Any idea on this now?

**Comment by iwasthesword** [comment]
You can add a new version to the `laragon\bin\php` folder and change the version whenever you want in the menu

**Comment by anburocky3** [comment]
> You can add a new version to the `laragon\bin\php` folder and change the version whenever you want in the menu

I want to have multiple PHP versions for multiple PHP projects.

```
for example: 
Project A works with PHP7.1
Project B works with PHP8.1
```

I want to run both at once.

**Comment by wnunezc** [comment]
> > You can add a new version to the `laragon\bin\php` folder and change the version whenever you want in the menu
> 
> I want to have multiple PHP versions for multiple PHP projects.
> 
> ```
> for example: 
> Project A works with PHP7.1
> Project B works with PHP8.1
> ```
> 
> I want to run both at once.

i find it but in spanish:  https://es.stackoverflow.com/a/432920/306965

**Comment by iwasthesword** [comment]
> > You can add a new version to the `laragon\bin\php` folder and change the version whenever you want in the menu
> 
> I want to have multiple PHP versions for multiple PHP projects.
> 
> ```
> for example: 
> Project A works with PHP7.1
> Project B works with PHP8.1
> ```
> 
> I want to run both at once.

I dont think that's possible. Maybe if you ran two portable versions but then you'd probably have issues with conflicting ports

**Comment by wnunezc** [comment]
> > > You can add a new version to the `laragon\bin\php` folder and change the version whenever you want in the menu
> > 
> > 
> > I want to have multiple PHP versions for multiple PHP projects.
> > ```
> > for example: 
> > Project A works with PHP7.1
> > Project B works with PHP8.1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I want to run both at once.
> 
> I dont think that's possible. Maybe if you ran two portable versions but then you'd probably have issues with conflicting ports

You are wrong, there are two ways to do it:
1. the basic one is through virtual host and use an application proxy that points to the version that you want to run... php is executed at the application level... it is apache that occupies port 80, not php...

2. use a reverse nginx proxy to point to local/docker/virtual machine server and port ... that run the version of php you need.

**Comment by reactmore** [comment]
> i find it but in spanish: https://es.stackoverflow.com/a/432920/306965

does this method work without problems? thanks for the reference

![MULTI_PHP](https://user-images.githubusercontent.com/64960434/201460002-ddd45d10-d325-41f7-85ec-490d8afc2614.jpg)

**Comment by bedoz** [comment]
I think that having this on menu could be the best feature to have for this quarter of 2023.
Not just with auto edit of apache or nginx config but also with a php alias for terminal when i want to open it by site, with additional menu voice "terminal start from" -> "site1", "site2", "site3", ... when i click on site1 it start terminal with site1 php version, and so on.

Follow this thread - checked!

**Comment by blockhunts** [comment]
I am also interested in this topic, I hope there will be a option to easily set the PHP version for each application.

**Comment by fxgourves** [comment]
> i find it but in spanish: https://es.stackoverflow.com/a/432920/306965

This solution works very well thanks !

**Comment by dangnhdev** [comment]
> i find it but in spanish: https://es.stackoverflow.com/a/432920/306965

How to do this with nginx?

**Comment by wnunezc** [comment]
@dangnhdev and all:

ok, to update the focus: the laragon stack has all the same features of a development server stack, only it includes a range of additional tools to have a programming-ready environment... which is nice and makes it easier to use. fact that if you have to make any changes to a project from a single download you can do everything...

Regarding the adjustments and features it has, they remain the same, if you know how to add new urls to nginx, simply modify the nginx configuration file... the same thing you do with Apache on the vhost...

I have even gone so far as to configure Laragon's nginx to function as a reverse-proxy at the same time that I use it as a local application-proxy, at the same time that I have modified it to point to the IP of virtual machines and/or Docker containers.

The general nginx/apache documentation is the same one you'll use with laragon... just be careful to disable creating new vhosts to avoid overwriting your changes...
